### PR TITLE
Fix word separation

### PR DESCRIPTION
**Pull request by ajalab**
**What this PR does / why we need it**:

This PR fixes the degradation issue that words which contain diacritics (e.g., `Städte`) are separated in word motions.

**Which issue(s) this PR fixes**

Two issues are reported in #3665:

1.  A string which contains both Latin alphabets and `vim.iskeyword` characters (e.g., `helloWorld(param1`) is wrongly regarded as a single word https://github.com/VSCodeVim/Vim/issues/3665#issue-430275917i.
1.  A string which contains both Latin ASCII alphabets and diacritics (e.g., `adbdémm`, `Städte`) is wrongly regarded as separate words https://github.com/VSCodeVim/Vim/issues/3665#issuecomment-481040030.

This PR fixes the second one. I've not been able to reproduce the original first problem...

**Special notes for your reviewer**:

- Auxiliary methods (`makeAsciiWordSegments` and `makeUnicodeWordSegments`) are removed.
- The behavior of the regex can be checked in https://regex101.com/r/X1agK6/1.

### CODE REVIEW

**Comment by ajalab** [comment]
Thanks to the issue report #3680, it was found that [Latin-1 symbols and punctuations](https://en.wikipedia.org/wiki/List_of_Unicode_characters#Latin-1_Supplement) were wrongly classified as word characters. I overlooked that the original Vim treats characters whose code point is less than 0x100 (both ASCII chars *and Latin-1 chars*) as a special case in [mbyte.c#L2879](https://github.com/vim/vim/blob/55d3bdbbe2bfc7a78b4aa17763788dbddf87cab0/src/mbyte.c#L2879). So I added Latin-1 symbols and punctuations (code point range: `0x00a1`-`0x00bf`) to the symbol table in https://github.com/VSCodeVim/Vim/pull/3667/commits/e831849a3d068a157ef4679de39564fc50180fbb.

This PR fixes #3680.

File: src/common/motion/position.ts [hunk 1]
```diff
@@ -946,10 +908,11 @@ export class Position extends vscode.Position {
       Hangul,
     }
 
+    // List of printable characters (code point intervals) and their character kinds.
+    // Latin alphabets (e.g., ASCII alphabets and numbers,  Latin-1 Supplement, European Latin) are excluded.
     // Imported from utf_class_buf in src/mbyte.c of Vim.
-    // Spelling alphabets are not listed here since they are covered as non-white letters.
-    // TODO(ajalab): add Emoji
-    const codePointRanges: [[number, number], CharKind][] = [
+    const table: [[number, number], CharKind][] = [
```
Comment (jpoon) [inline src/common/motion/position.ts]: nit: rename this a more descriptive name.

Comment (ajalab) [reply src/common/motion/position.ts]: how about `symbolTable`?

Comment (jpoon) [reply src/common/motion/position.ts]: Works for me.

**Pull request merged**